pkg/rhcos/ami.go

```diff
@@ -10,17 +10,11 @@ import (
 	"github.com/pkg/errors"
 )
 
-const (
-	// DefaultChannel is the default RHCOS channel for the cluster.
-	DefaultChannel = "tested"
-)
-
 // AMI calculates a Red Hat CoreOS AMI.
+// TODO: Use the new RHCOS build pipeline to discover AMIs. The pipeline isn't
+//       currently exposed publicly and CI isn't on the VPN, so this cannot be
+//       updated without breaking the tests.
```

Comment:
This is only workable if we have tge same internal and external release cadence.  See discussion in #281 and openshift/os#353.

Reply:
> How hard would it be to put some of this data in the pull secret though or so?

What data?  A URI with an initial entry-point for this discovery?

Reply:
> What data? A URI with an initial entry-point for this discovery?

I'd say all the release metadata - including the `meta.json` fetched by this PR, which includes AMI ids, along with link to qcow2 etc.

Hmm.  Or maybe the pull secret grants access to a server running on api.ci or wherever that hands out the metadata.

Reply:
Well, one thing this circles back to is...we could stick all of the relevant metadata in a container, which the pull secret already grants access to of course.

Would having this be in a container be ergonomic for the installer?  It'd depend on `oc adm release extract`?

Reply:
/cc @smarterclayton

Reply:
> Would having this be in a container be ergonomic for the installer? It'd depend on oc adm release extract?

We currently do not require image-pulling logic or `oc` on the install host.  I'd rather not add those dependencies, but if we add them this approach would work.

I'm more in favor of your ["pull secret grants access to a server running on api.ci or wherever that hands out the metadata"][1].

[1]: https://github.com/openshift/installer/pull/554#discussion_r229019273

Reply:
Middle ground approach: pull secret is just used to grab container *metadata*, retrieved via e.g. `skopeo inspect`.  Installer takes `skopeo` as external dependency or vendors it?  This approach doesn't require having a whole container storage setup.

Reply:
This all sounds far too complex for what we are after. A few notes on the discussion:

  - We shouldn't depend on the URL remaining unknown in order to protect the RHCOS images. Hiding this in a container behind a pull-secret doesn't buy much.
  - The RHCOS image reference should come from the CVO manifest payload (just like all of the other resources). We need to wait until MAO can create masters before we can make this jump since I don't want the installer parsing through the CVO manifest payload.

A lot of this is temporary scaffolding, so I'd rather we focus the discussion on the longer-term vision rather than this code which we know we are going to throw away in a few weeks.

Reply:
> The RHCOS image reference should come from the CVO manifest payload (just like all of the other resources). We need to wait until MAO can create masters before we can make this jump since I don't want the installer parsing through the CVO manifest payload.

What approach would you use for the always-pre-MAO bootstrap (which needs an AMI too ;)?

Reply:
> We shouldn't depend on the URL remaining unknown in order to protect the RHCOS images. Hiding this in a container behind a pull-secret doesn't buy much.

True long term.

> The RHCOS image reference should come from the CVO manifest payload

Agree.  What's fuzzy for me is the dependencies required for that.  It also forces some deep issues like...when we branch OKD, are we going to try to maintain a separate OS stream for that?  Or do we just keep the same OS and ship updated kubelets too?  Or a mix where we ship the same "base OS", and just pull different kubelets?  We've talked about this in the past, and what hurts my head is that the complexity jumps up when one notes we have two major version streams for RHCOS today.

Reply:
Confirmed.